### ISSUE TITLE
Records fail on optional fields

### ISSUE THREAD

**Issue by jsinclair-mckesson** [eclipse-ee4j/yasson#536]
**Describe the bug**
The recently added support for java records only works if every JSON object contains every possible field.  The vast majority of JSON in use contains optional fields that may not be present.  This causes an exception (e.g. "javax.json.bind.JsonbException: JsonbCreator parameter model is missing in json document.") when deserializing the document.

**To Reproduce**
Add the following (slightly adjusted) code to the existing unit tests.
```java
public record Car(String type, String model, String color) {} // properties are optional, and often not present
private static final Jsonb serializer = Jsonbs.defaultJsonb; 
private static final Jsonb deSerializer = Jsonbs.defaultJsonb; 
@Test
public void testRecordJsonbMissingProperties() {
    Car carMissingColor = new Car("Tesla", "Model 3", null);
    Car carMissingModel = new Car("Toyota", null, "Metallic Silver");
    Car carOnlyType = new Car("Volvo", null, null);
    String expectedMissingColor = """
    {
        "type": "Tesla",
        "model": "Model 3"
    }
    """;
    String expectedMissingModel = """
    {
        "type": "Toyota",
        "color": "Metallic Silver"
    }
    """;
    String expectedOnlyType = """
    {
        "type": "Volvo"
    }
    """;
    // All three cases below fail with exception on deserialize (fromJson)
    String jsonMissingColor = serializer.toJson(carMissingColor);
    Car deserializedMissingColor = deSerializer.fromJson(expectedMissingColor, Car.class);
    assertEquals(expectedMissingColor, json);
    assertEquals(carMissingColor, deserializedMissingColor);

    String jsonMissingModel = serializer.toJson(carMissingModel);
    Car deserializedMissingModel = deSerializer.fromJson(expectedMissingModel, Car.class);
    assertEquals(expectedMissingModel, json);
    assertEquals(carMissingModel, deserializedMissingModel);

    String jsonOnlyType = serializer.toJson(carOnlyType);
    Car deserializedOnlyType = deSerializer.fromJson(expectedOnlyType, Car.class);
    assertEquals(expectedOnlyType, json);
    assertEquals(carOnlyType, deserializedOnlyType);
}
```

**Expected behavior**
The object should serialize to JSON with, or without, the null property filled in as "null" based only on whether `withNullValues(boolean)` is set on the JsonBuilder.
The object should deserialize from JSON with null for any parameter to the record that is not present in the JSON being parsed.

**System information:**
 - OS: Any
 - Java Version: 17
 - Yasson Version: 1.0.11 and 2.0.4

**Additional context**
The goal is to permit using record classes to accept the JSON payload sent to a ReST based microservice built on Microprofile.

**Comment by Verdent** [comment]
Hi @jsinclair-mckesson , currently records in Yasson are using the same approach as if the constructor is annotated with `@JsonbCreator`. In version 1.x and 2.x there was no mechanism for default values to be passed if property is missing. We will be releasing 3.0.0-RC1 version of the Yasson very soon. This version does not have this restriction anymore since this requirement for the properties to be present in the JSON document has been relaxed in the spec. 

It might make sense to change it for records in 2.x and 1.x also.

**Comment by jsinclair-mckesson** [comment]
Thank you for the update.  I noticed the approach, which is why I provided some unit test code to demonstrate the issue. We can't use the `@JsonbCreator` annotation anywhere for similar reasons; it breaks on optional fields.
The sooner we can have records supported fully as JSON proxies, without failing on optional fields, the better.
Right now we have to write, on average, several hundred lines of unnecessary boilerplate code per JSON object to implement accessors, mutators, and other methods that we prefer not exist. We have scripting to create objects from lists of fields, but it's still many thousands of lines of code we could eliminate if records were supported more fully.

We would definitely need this backported to 1.x and 2.x, if at all possible, because the microprofile platforms we must use are not ready to migrate to the 3.x versions (quarkus, for instance, is still on 1.x for a while).
